script/validate-categories/validate.rb

```diff
@@ -0,0 +1,37 @@
+require 'json'
+
+settings = JSON.parse(File.read('./settings.json'))
+folders = settings['folders']
+@allowed_categories = settings['allowed_categories']
+
+def validateCategories(categories)
+    return categories.nil? || (categories.is_a?(Array) && ! categories.select{|l| @allowed_categories.detect{|p| p.casecmp(l) == 0 } }.empty? )
+end
+
+result = []
+for folder in folders
+    files = Dir.entries(folder).select {|entry| File.file?(File.join(folder, entry)) && (File.extname(entry) == ".yaml" || File.extname(entry) == ".yml") }
```

Comment:
Do we need to co-relate with workflow yml file? Or just simply look through all properties file?

Reply:
The co-relation is taken care by the validate-data script.

Reply:
No, I meant why do we need to first find the workflow file name and then look for properties.json for that? We can directly iterate through all properties.json and validate category.